Src/FluentAssertions/CallerIdentification/MultiLineCommentParsingStrategy.cs

```diff
@@ -11,7 +11,7 @@ public ParsingState Parse(char symbol, StringBuilder statement)
     {
         if (isCommentContext)
         {
-            var isEndOfMultilineComment = symbol == '/' && commentContextPreviousChar == '*';
+            var isEndOfMultilineComment = symbol is '/' && commentContextPreviousChar is '*';
```

Comment:
Why changing `==` to `is`? What's the added value? Or is this a matter of taste?

Reply:
I think @jnyrup first [sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+B6DACAvL7AdwEsA7AZwFgAocgFygFcw7sBlBsgcxoG8bsg7AAEAzNmAQIAG2wBRAI6MAhtPIAKYQEYADNnpQAlHgB8+hgH4AdABkYpLnQAWeAjoDcAoV8FZsTujoAB3IQLC5iZ0ZgK0gAWwwAEwg6Uhg6DCgIcmkAT1IMYnJyRhhyDCQAVh0ATi0fEXFJGWwABWVAmChSAFkOsCdNXXMjUxHrOwdnbCLsDxoAXxoaPwByItWiMipaBmZWAGEnZSg+BrEJKVlFFTV1AZOLbDBjXDMwV2xVgEFVz2ohNhzk0rm0OnQur1+oMHlAni8xh9Zj8/otltQ/AA5CDYAD6YGkMBOuK2pDSp12TBY2AAkqQ6GcAUILs1rkpVBoyHQnsRXmZiJ8tP9AcDLi12p1un06AN1FyeXyZjNyNghWjqDQgA===) analysis should give a hint :)

Reply:
Actually, it doesn't the `String`/`Equals` version looks simpler (and supposedly would be faster).

Reply:
`commentContextPreviousChar` is a `char?` which is the second example in the SharpLab example.

For a plain `char` like `symbol` there's no difference in the generated code.

This [SharpLab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+BLAdgFxlLAQwBsACAAQCZSBZAV2JwwBlsYBhCAWy5lwAVCUAM7YA5gGUcUQnjEBPALAAoAN4rSm0gAcoGAG6yYpYBAhkMwzjz45OuGAhwBuDVt0GjpMAAshAfm9uXlx7PCd+WH0MCDorPyhXZS1SN01yAGYTMzIASWEAUSwAEwB5ADN6RgxiNmsQnAB9UuJigApfIVJheS5TYgBKNNJ1ZJStDHLSNst62zDHHCGx8ZHh1fSAdm7e/tIAXn3SAHIAemPSADJLoJtQiAcIqJi49gSDo+OAKmOkjdIAL7rLTArakcokYQwP5aIErVLwzLZcykfJFMqVBhMWpYDjBWyNAByMAA7h13j0+uZlilRv9JtNZvj7o8lqC1vD/uRtpS9pYTucrjdIHc7A9wjhIjBorF4l1+d9fuy4f92dzwZDocM4QCgA==)

Show the difference for a simplified version of this method.

Reply:
So `is` does result in a slightly faster implementation. Not sure I understand why there's a difference in the generated code though.

Reply:
> Not sure I understand why there's a difference in the generated code though.

Normally I would guess that a [peephole optimization](https://en.wikipedia.org/wiki/Peephole_optimization) simply hasn't been implemented, but since it's the newer syntax that generates better code, I would put my money on that because an [optimizing compiler](https://en.wikipedia.org/wiki/Optimizing_compiler) might need to stay conservative to preserve correctness, there is some slight difference between the two operators which it hard to impossible to optimize in the general case.